tests/test_tr_link.py

```diff
@@ -0,0 +1,139 @@
+from pathlib import Path
+
+import importlib.util
+import sys
+
+
+def import_from_path(path, name):
+    spec = importlib.util.spec_from_file_location(name, path)
+    module = importlib.util.module_from_spec(spec)
+    sys.modules[name] = module
+    spec.loader.exec_module(module)
+    return module
+
+
+# (!) We need to import the 'tr_link' method from the local module,
+# otherwise the method from the installed 'sphinxcontrib' package will be used.
+local_functions_module = import_from_path(

```

Comment:
On my machine a `from sphinxcontrib.test_reports.functions import tr_link` was enough, after I did:

```
pip install .
pip install -r test-requirements.txt
pip install sphinxcontrib-plantuml  # a missing dep in the sphinx-needs config
```

So I guess we do not need this kind of import here.

Reply:
I am using a virtual environment.
I did following:
* create `.venv` directory in the repository
* run `pipenv install -r test-requirements.txt`

Executing the tests with `pytest` fail because `sphinx` is not installed. So the dependencies in the `test-requirements.txt` are not enough to execute the tests.

I run `pipenv install -r doc-requirements.txt`.
All tests now pass.

I removed the ugly `import_from_path ` and just added `from sphinxcontrib.test_reports.functions import tr_link`. 
Now the `test_tr_link.py` tests are failing because the `tr_link` from the `sphinxcontrib` is used. I raised an exception in the local `tr_link` to make sure this is actually not used. The exception was not raised and the tests are failing.

So my problem is caused by installing the doc dependencies as these contain `sphinx-test-reports>=0.3.3`.

I am not familiar with `nox` so I did check why this is not a problem on CI.

There are two things to clarify:

* which steps a developer shall do locally to have a running test environment
* how to make sure that the correct `tr_link` method is used also when `sphinxcontrib.test_reports` is available in the Python environment.

Reply:
Thanks for the insights into your setup, I guess `pip install .` or similar is missing to install the locally checked-out Sphinx-Test-Report package. Then the correct `tr_link` should be taken for tests.